### PR TITLE
[WIP]Out-of-order: Extending Appendable interface to include hints

### PR DESCRIPTION
**Pull request by Vanshikav123**

### CODE REVIEW

File: storage/fanout.go [hunk 1]
```diff
@@ -118,11 +118,11 @@ func (f *fanout) ChunkQuerier(mint, maxt int64) (ChunkQuerier, error) {
 	return NewMergeChunkQuerier([]ChunkQuerier{primary}, secondaries, NewCompactingChunkSeriesMerger(ChainedSeriesMerge)), nil
 }
 
-func (f *fanout) Appender(ctx context.Context) Appender {
-	primary := f.primary.Appender(ctx)
+func (f *fanout) Appender(ctx context.Context, hints *AppendHints) Appender {
+	primary := f.primary.Appender(ctx, nil)
```
Comment (bboreham) [inline storage/fanout.go:122]: Should this pass `hints` ?

File: cmd/prometheus/main.go [hunk 2]
```diff
@@ -1563,9 +1563,9 @@ func (s *readyStorage) ExemplarQuerier(ctx context.Context) (storage.ExemplarQue
 }
 
 // Appender implements the Storage interface.
-func (s *readyStorage) Appender(ctx context.Context) storage.Appender {
+func (s *readyStorage) Appender(ctx context.Context, hints *storage.AppendHints) storage.Appender {
 	if x := s.get(); x != nil {
-		return x.Appender(ctx)
+		return x.Appender(ctx, nil)
```
Comment (bboreham) [inline cmd/prometheus/main.go:1568]: Should this pass `hints` ?

File: storage/interface.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ type Appendable interface {
 	// Appender returns a new appender for the storage. The implementation
 	// can choose whether or not to use the context, for deadlines or to check
 	// for errors.
-	Appender(ctx context.Context) Appender
+	Appender(ctx context.Context, hints *AppendHints) Appender
```
Comment (bboreham) [inline storage/interface.go:65]: Add something to the comment explaining the new parameter.

**Comment by jesusvazquez** [comment]
I'll close this PR since we've discarded this proposal.

**Pull request closed without merging**